lib/get-exporter.js

```diff
@@ -0,0 +1,113 @@
+const paginateRequest = require('./paginate-request');
+const inquirer = require('inquirer');
+const {
+  loadConfig,
+  getShallowStatus,
+  logResult,
+  logProcessing,
+  logError,
+  mapById,
+  plural
+} = require('./utils');
+const {
+  remove,
+  writeFile,
+  ensureDir
+} = require('fs-extra');
+const { values, isEmpty } = require('omnibelt');
+const allSettledSerial = require('./all-settled-serial-p');
+const { rollbarLog } = require('./rollbar');
+
+const getExporter = ({
+  apiType, commandType, localStatusParams, remoteStatusParams, getData
+}) => {
+  const exporter = async (pattern, command = {}, loadedConfig) => {
+    let didExport = false;
+    const { apiToken, applicationId, api } = loadedConfig ? loadedConfig : await loadConfig();
+    if (!apiToken || !applicationId) { return; }
+    await ensureDir(commandType);
+    let items;
+    try {
+      items = await paginateRequest(api[apiType].get, { applicationId });
+    } catch (e) {
+      return logError(e);
+    }
+    const itemsById = mapById(items);
+    const { remoteStatusByFile, localStatusByFile } = await getShallowStatus({
+      items,
+      remoteStatusParams,
+      localStatusParams,
+      dir: commandType,
+      pattern
+    });
+    if (command.dryRun) {
+      logResult('DRY RUN');
+    }
+    if (isEmpty(remoteStatusByFile)) {
+      if (!pattern) {
+        return logResult('Missing', `No ${plural(commandType)} found to export.`, 'yellow');
+      } else {
+        return logResult('No Matches', `No ${plural(commandType)} found that match this pattern ${pattern}`, 'yellow');
+      }
+    }
+    const downloadResults = await allSettledSerial(async (remoteStatus) => {
+      logProcessing(remoteStatus.file);
+      if (!command.force) {
+        const conflict = localStatusByFile.has(remoteStatus.file) && remoteStatus.status === 'found';
+        if (conflict) {
+          const { handleConflict } = await inquirer.prompt([{
+            name: 'handleConflict',
+            type: 'confirm',
+            message: `The file ${remoteStatus.file} already exists, do you want to overwrite it?`
+          }]);
+          if (!handleConflict) {
+            return logResult('skipped', remoteStatus.file, 'yellow');
+          }
+        }
+      }
+      if (remoteStatus.status === 'missing' && !command.dryRun) {
+        const { canDelete } = await inquirer.prompt([{
+          name: 'canDelete',
+          type: 'confirm',
+          message: `The file ${remoteStatus.file} is missing remotely, do you want to delete it locally?`
```

Comment:
this printed out as `The file true is missing remotely, do you want to delete it locally` instead of actually using the file name

Reply:
and then it did `Error An Error occurred when trying to delete file true with the message rimraf: path should be a string`